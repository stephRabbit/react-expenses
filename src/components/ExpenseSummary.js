import React from 'react';
import { connect } from 'react-redux';
import selectExpense from '../selectors/expenses';
import getTotalExpense from '../selectors/expense-total';
import numeral from 'numeral';

export const ExpenseSummary = ({ expenseCount, expenseTotal }) => {
  const expenseText = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedTotal = numeral(expenseTotal / 100).format('$0,0.00');
  return (
    <div>
      <h2>
        Viewing {expenseCount} {expenseText} totalling { formattedTotal }
      </h2>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpense = selectExpense(state.expenses, state.filters);

  return {
    expenseCount: visibleExpense.length,
    expenseTotal: getTotalExpense(visibleExpense)
  };
};

export default connect(mapStateToProps, null)(ExpenseSummary);