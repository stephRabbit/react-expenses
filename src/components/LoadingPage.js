import React from 'react';

const LoadingPage = () => (
  <div className="loader">
    <div className="loader__item">
      <div className="loader__rect1"></div>
      <div className="loader__rect2"></div>
      <div className="loader__rect3"></div>
      <div className="loader__rect4"></div>
      <div className="loader__rect5"></div>
    </div>
  </div>
);

export default LoadingPage;