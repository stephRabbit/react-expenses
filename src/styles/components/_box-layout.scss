.box-layout {
  align-items: center;
  background-color: $green;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.box-layout__box {
  background: fade-out($color: #fff, $amount: .15);
  -webkit-box-shadow: 0 7px 15px rgba(0,0,0,.2);
  box-shadow: 0 7px 15px rgba(0,0,0,.2);
  text-align: center;
  padding: $l-size $m-size;
  width: 35rem;
}

.box-layout__title {
  line-height: 1;
  margin: 0 0 $m-size 0;
}