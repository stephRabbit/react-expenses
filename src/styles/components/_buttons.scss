.btn {
  background: white;
  border: 0;
  border-radius: 3px;
  color: white;
  display: inline-block;
  font-size: $fs-large;
  padding: 1rem 2rem;
  text-decoration: none;
}

.btn--small {
  font-size: $fs-small;
}

.btn--green {
  color: white;
  background-color: $green;
}

.btn--red {
  color: white;
  background-color: $red;
}

.btn--fade {
  background: fade-out($color: #fff, $amount: .70);
}

.btn--lifted {
  -webkit-box-shadow: 0 7px 15px rgba(0,0,0,.2);
  box-shadow: 0 7px 15px rgba(0,0,0,.2);
}

