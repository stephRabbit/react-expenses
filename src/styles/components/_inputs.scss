.text-input {
  border: 1px solid #cacccd;
  height: 50px;
  padding: $s-size;
  font-size: $fs-large;
}

.select-input {
  @extend .text-input;
}

.textarea-input {
  @extend .text-input;
  height: 10rem;
}