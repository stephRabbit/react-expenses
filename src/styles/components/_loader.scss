.loader {
  background: $light-green;
  align-items: center;
  height: 100vh;
  justify-content: center;
  display: flex;
  width: 100vw;
}

.loader__item {
  font-size: 10px;
  height: 40px;
  width: 50px;
  text-align: center;

  > div {
    background-color: $green;
    display: inline-block;
    height: 100%;
    margin-right: 3px;
    width: 6px;

    -webkit-animation: sk-stretchdelay 1.2s infinite ease-in-out;
    animation: sk-stretchdelay 1.2s infinite ease-in-out;
  }

  .loader__rect2 {
    -webkit-animation-delay: -1.1s;
    animation-delay: -1.1s;
  }

  .loader__rect3 {
    -webkit-animation-delay: -1.0s;
    animation-delay: -1.0s;
  }

  .loader__rect4 {
    -webkit-animation-delay: -0.9s;
    animation-delay: -0.9s;
  }

  .loader__rect5 {
    -webkit-animation-delay: -0.8s;
    animation-delay: -0.8s;
  }
}

@-webkit-keyframes sk-stretchdelay {
  0%, 40%, 100% { -webkit-transform: scaleY(0.4) }
  20% { -webkit-transform: scaleY(1.0) }
}

@keyframes sk-stretchdelay {
  0%, 40%, 100% {
    transform: scaleY(0.4);
    -webkit-transform: scaleY(0.4);
  }  20% {
    transform: scaleY(1.0);
    -webkit-transform: scaleY(1.0);
  }
}