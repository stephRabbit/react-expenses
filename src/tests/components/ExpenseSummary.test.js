import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should render ExpenseSummary with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={100} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={3} expenseTotal={100000} />);
  expect(wrapper).toMatchSnapshot();
});