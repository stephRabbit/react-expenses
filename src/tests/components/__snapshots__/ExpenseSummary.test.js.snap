// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $1.00
  </h2>
</div>
`;

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    3
     
    expenses
     totalling 
    $1,000.00
  </h2>
</div>
`;
